File: itcloud/src/main/java/com/itinfo/itcloud/controller/LoginController.java
package com.itinfo.itcloud.controller;

import com.itinfo.itcloud.service.SystemPropertiesService;
import com.itinfo.itcloud.service.UserService;
import com.itinfo.itcloud.ovirt.ConnectionService;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

@Controller("LoginController")
@Slf4j
public class LoginController {

    @Autowired
    private ConnectionService ovirtConnection;

    @Autowired
    private SystemPropertiesService systemPropertiesService;

    @Autowired
    private UserService userService;


    public LoginController(){}

    // 로그인 페이지
    @GetMapping("/login")
    public String login(){
        log.info("-----login");
        return "login/loginpage";
    }

    /* name이 null이면 login이 나오는데 로그인이 되지 않으면 이상한 페이지가 뜸 *//*
    // login_check 명으로 dashboard가 뜸
    @PostMapping("/login_check")
    public String login_check(LoginDTO loginDTO, Model model, HttpServletRequest httpServletRequest) {
        String name = userService.login(loginDTO);
        String id = loginDTO.getId();

        SystemPropertiesVO systemPropertiesVO = this.systemPropertiesService.searchSystemProperties();
        if(systemPropertiesVO.getId() != "" && systemPropertiesVO.getPassword() != "" && systemPropertiesVO.getIp() != ""){
            return "redirect:dashboard";
        }else{
            return "redirect:login/loginpage";
        }

    }



    // ovirt 연결 테스트
    @GetMapping("/scopetest")
    public String scopeTest(Model model, HttpServletRequest request) {
        model.addAttribute("uid", this.ovirtConnection.getUid());
        return "scopetest";
    }


    // db연결 확인 테스트
    @GetMapping("/userCount")
    public String goTestPage(Model model, HttpServletRequest request) {
        int result = 0;
        result = userService.getTestValue();

        model.addAttribute("userCount", result);
        return "user";
    }

    // json 형식
    @GetMapping("/testuser")
    @ResponseBody
    public List<UserVO> list(){
        List<UserVO> users = null;
        try {
            users = userService.allUsers();
        }catch (Exception e){
            e.printStackTrace();
        }
        return users;
    }
*/

}

File: itcloud/src/main/java/com/itinfo/itcloud/controller/auth/LoginController.kt
package com.itinfo.itcloud.controller.auth

import com.itinfo.common.LoggerDelegate
import com.itinfo.itcloud.controller.BaseController
import com.itinfo.itcloud.model.auth.UserVo

import com.itinfo.itcloud.service.auth.ItOvirtUserService
import io.swagger.annotations.Api
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@Api(tags = ["login"])
@RequestMapping("/api/v1/auth/")
class LoginController: BaseController() {
	@Autowired private lateinit var ovirtUser: ItOvirtUserService
	// name이 null이면 login이 나오는데 로그인이 되지 않으면 이상한 페이지가 뜸
	// login_check 명으로 dashboard가 뜸
/*
	@PostMapping
	fun login_check(loginDTO: LoginDTO, model: Model?, httpServletRequest: HttpServletRequest?): String {
		val name: String = userService.login(loginDTO)
		val id: String = loginDTO.getId()

		val systemPropertiesVO: SystemPropertiesVO = this.systemPropertiesService.searchSystemProperties()
		return if (systemPropertiesVO.getId() !== "" && systemPropertiesVO.getPassword() !== "" && systemPropertiesVO.getIp() !== "") {
			"redirect:dashboard"
		} else {
			"redirect:login/loginpage"
		}
	}
*/
	@GetMapping("users")
	fun findAllOvirtUsers(): ResponseEntity<List<UserVo>> {
		log.info("findAllOvirtUsers ... ")
		val res: List<UserVo> = ovirtUser.findAll()
		return ResponseEntity.ok(res)
	}

	companion object {
		private val log by LoggerDelegate()
	}
}